src/tlo/methods/wasting.py

```diff
@@ -1,41 +1,1298 @@
+        modules would read a particular file within here.
+        :return:
+        """
+        # Update parameters from the resource dataframe
+        # Read parameters from the resource file
+        self.load_parameters_from_dataframe(
+            pd.read_excel(Path(self.resourcefilepath) / 'ResourceFile_Wasting.xlsx', sheet_name='Parameter_values_AM'))
+
+        # Register wasting symptom(weight loss) in Symptoms Manager
+        self.sim.modules['SymptomManager'].register_symptom(Symptom(name=self.wasting_symptom))
 
     def initialise_population(self, population):
+        """
+        Set our property values for the initial population. This method is called by the simulation when creating
+        the initial population, and is responsible for assigning initial values, for every individual,
+        of those properties 'owned' by this module, i.e. those declared in the PROPERTIES dictionary above.
+
+        :param population:
+        :return:
+        """
         df = population.props
-        df.loc[df.is_alive, 'un_clinical_acute_malnutrition'] = 'well'
+
+        # Set initial properties
         df.loc[df.is_alive, 'un_ever_wasted'] = False
+        df.loc[df.is_alive, 'un_WHZ_category'] = 'WHZ>=-2'  # not undernourished
+        df.loc[df.is_alive, 'un_clinical_acute_malnutrition'] = 'well'
+        df.loc[df.is_alive, 'un_last_wasting_date_of_onset'] = pd.NaT
+        df.loc[df.is_alive, 'un_acute_malnutrition_tx_start_date'] = pd.NaT
+        df.loc[df.is_alive, 'un_sam_death_date'] = pd.NaT
+        df.loc[df.is_alive, 'un_am_bilateral_oedema'] = False
+        df.loc[df.is_alive, 'un_am_MUAC_category'] = '>=125mm'
+        df.loc[df.is_alive, 'un_am_treatment_type'] = 'not_applicable'
+
+        # initialise wasting linear models.
+        self.wasting_models = WastingModels(self)
+
+        # Assign wasting categories in young children at initiation
+        for low_bound_mos, high_bound_mos in [(0, 5), (6, 11), (12, 23), (24, 35), (36, 47), (48, 59)]:  # in months
+            low_bound_age_in_years = low_bound_mos / 12.0
+            high_bound_age_in_years = (1 + high_bound_mos) / 12.0
+            # linear model external variables
+            agegp = f'{low_bound_mos}_{high_bound_mos}mo'
+            mask = (df.is_alive & df.age_exact_years.between(low_bound_age_in_years, high_bound_age_in_years,
+                                                             inclusive='left'))
+            prevalence_of_wasting = self.wasting_models.get_wasting_prevalence(agegp=agegp).predict(df.loc[mask])
+
+            # categorize into moderate (-3<=WHZ<-2) or severe (WHZ<-3) wasting
+            wasted = self.rng.random_sample(len(prevalence_of_wasting)) < prevalence_of_wasting
+            for idx in prevalence_of_wasting.index[wasted]:
+                probability_of_severe = self.get_probs_or_odds_wasting(agegp=agegp)
+                wasted_category = self.rng.choice(['WHZ<-3', '-3<=WHZ<-2'], p=[probability_of_severe,
```

Comment:
do we want same wasting category applied to all individuals of the same age group? I thought this should be applied  on an individual basis i.e. different individuals may have different wasting categories even if they're of the same age group?

Reply:
Yes, it should be applied on an individual basis, but `.apply` can be used to apply the same function to each column or row, right? E.g. to set a values within one column can be done for all lines based on values in other column(s) in corresponding lines. 

So you could apply the fnc `choice` to each wasted individual to get for each of them an individualy based wasted category, and similarly set the attributes for each of them.

_But we could maybe leave this for now as it is and note somewhere that this may be improved later._